### ISSUE TITLE
[Preflight Checks] Check if distribution is supported

### ISSUE THREAD

**Issue by prppedro** [replicated-collab/rocketchat-kots#21]
As Josh pointed out, maybe we want to restrict the supported platforms. Our software, itself, only runs on Docker based systems. 

cf. https://troubleshoot.sh/docs/analyze/distribution/

**Comment by jdewinne** [comment]
@prppedro Is it docker based or containerd for the runtime? Or both?
Also I think this refers more to the k8s distro like checking for GKE, EKS, etc.

**Comment by prppedro** [comment]
@jdewinne, I think it's docker based. But I'm not really sure, ATM. 

As for the GKE, EKS, AKS, etc... Maybe we need to test them individually at each provider/distro. For now, I know for a fact it works both on GKE and k3d.

**Comment by prppedro** [comment]
Yeah, it seems to depend on Docker, indeed. I'll, maybe, run some tests for other distributions. As for the cloud solutions, I'll have to ask the team. I know for a fact it was deployed to AWS in the past, but it was quite some time ago.
